java/src/jmri/jmrix/tams/TamsSensorManager.java

```diff
@@ -97,51 +121,55 @@ protected Sensor createNewSensor(@Nonnull String systemName, String userName) th
                     sensorList.put(channel, s);
                 }
             } catch (NumberFormatException ex) {
-                throw new IllegalArgumentException("Unable to convert " +  // NOI18N
-                        systemName.substring(getSystemPrefix().length() + 1) +
-                        " into the Module and port format of nn:xx"); // NOI18N
+                log.error("Unable to convert {} into the Module and port format of nn:xx", curAddress);
+                //return null;
             }
             if ((board * 2) > maxSE) {//Check if newly defined board number is higher than what we know
                 maxSE = board * 2;//adjust xSE and inform Tams MC
-                log.debug("Changed xSE to {}", maxSE);
-                TamsMessage tm = new TamsMessage("xSE " + Integer.toString(maxSE));
+                //log.info("Changed xSE to " + maxSE);
+                tm = new TamsMessage("xSE " + Integer.toString(maxSE));
                 tm.setBinary(false);
                 tm.setReplyType('S');
                 tc.sendTamsMessage(tm, this);
+                tmq.add(tm);
+                //no need to add a message for this board as the polling process will capture all board anyway
             }
         }
-        //Probably sending the status check 16 times but should work...
-        //Get initial status of sensors
-        TamsMessage tm = TamsMessage.setXSensOff(); //force report from sensors with at least 1 port set
-        tc.sendTamsMessage(tm, this);
-        tm = TamsMessage.getXEvtSen(); //reports only sensors with changed states
-        tc.sendTamsMessage(tm, this);
-        log.debug("Returning this sensor: {}", s.toString());
+        //log.info("Returning this sensor: " + s.toString());
         return s;
     }
 
     @Override
-    @Nonnull
-    public String createSystemName(@Nonnull String curAddress, @Nonnull String prefix) throws JmriException {
+    public String createSystemName(String curAddress, String prefix) throws JmriException {
         if (!curAddress.contains(":")) {
-            throw new JmriException("Hardware Address passed should be past in the form 'Module:port', was " + curAddress);
+            log.error("Unable to convert {} into the Module and port format of nn:xx", curAddress);
+            JOptionPane.showMessageDialog(null, Bundle.getMessage("WarningModuleAddress"),
```

Comment:
Hi @janboen , as #createSystemName could well be used by headless installs, prefereable to just throw the Exception ( perhaps the Bundle text is preferred? ).
#createSystemName is a relatively internal JMRI method, so any code using this should handle the Exception, and display the reason for the exception as appropriate ( which may include displaying a dialogue box ).

Reply:
I further developed on code from 2009 from Kevin Dickerson and I'm not a real java dev so what exactly do you want me to do?
Must this be done for the fix to be part of 4.25.4?

Reply:
The problem is that you have added the `showMessageDialog()` call. If you remove that and restore the original `throw JmriException`, then the problem is resolved.

In general, it's good to show the user a message dialog, but it only works if we know that the user is not running JMRI headless. In this particular case, I agree with @icklesteve that it's better to let the caller deal with the problem than showing a message to the user here.